backend/src/workers/tasks.py

```
@@ -234,122 +245,162 @@ async def _execute_agent_stream(
-                    "thread_id": thread_id,
-                    "error": str(e),
-                },
-            )
-
-    return {"status": "complete", "stream_key": stream_key}
+        return {"status": "complete", "stream_key": stream_key}
+    finally:
+        if daytona_sandbox is not None:
+            try:
+                daytona_sandbox.stop()
+            except Exception as exc:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)